tests/extract_configs/fixtures.py

```diff
@@ -0,0 +1,68 @@
+from creator.analyses.file_types import FILE_TYPES
+from creator.studies.data_generator.study_generator import StudyGenerator
+
+sg = StudyGenerator()
+sg._create_dfs()
+
+
+class MockAnalysis:
+    def __init__(self, columns):
+        self.columns = [{"name": c} for c in columns]
+
+
+class MockFile:
+    def __init__(self, file_type):
+        self.file_type = file_type
+
+
+class MockVersion:
+    def __init__(self, valid_types=(), columns=(), file_type=None):
+        self.analysis = MockAnalysis(columns)
+        self.valid_types = set(valid_types)
+        self.root_file = MockFile(file_type)
+
+
+def make_template_df(file_type):
+    """
+    Return a DataFrame with content that conforms to one of the templated
+    file types
+    """
+    dfs = {
+        "PDA": sg.dataframes["bio_manifest.tsv"],
+        "PTC": sg.dataframes["bio_manifest.tsv"],
+        "PTO": sg.dataframes["bio_manifest.tsv"],
+        "SAM": sg.dataframes["bio_manifest.tsv"],
+        "FTR": sg.dataframes["bio_manifest.tsv"],
+        "FCM": sg.dataframes["bio_manifest.tsv"],
+        "SAM": sg.dataframes["bio_manifest.tsv"],
+        "S3S": sg.dataframes["s3_source_gf_manifest.tsv"],
+        "GWO": sg.dataframes["gwo_manifest.tsv"],
+    }
+    assert set(dfs.keys()) <= set(FILE_TYPES.keys())
+
+    # Check all dfs conform to file type templates
+    for ft, df in dfs.items():
+        file_type_obj = FILE_TYPES[ft]
+        missing = []
+        for req_col in file_type_obj["required_columns"]:
+            if req_col not in df.columns:
+                missing.append(req_col)
+        if missing:
+            raise Exception(
+                f"Test DataFrame for file type {ft} is missing required "
+                f"columns: {missing}. Please modify the DataFrame "
+                "builders in StudyGenerator._df_creators, which can be "
+                "found in creator.studies.data_generator.study_generator"
+            )
```

Comment:
Do we need to run this part of the test for each different `file_type`? Maybe it could be separate?

Reply:
I could move this into another test so it doesn't check every file type every time you use this function